clang/tools/driver/CacheLauncherMode.cpp

```diff
@@ -162,12 +163,26 @@ static void addLauncherArgs(SmallVectorImpl<const char *> &Args,
       Args.push_back(Saver.save("-fdepscan-prefix-map=" + PrefixMap).data());
     }
   }
-  if (const char *ServicePath =
-          ::getenv("LLVM_CACHE_REMOTE_SERVICE_SOCKET_PATH")) {
+
+  const char *ServicePath = ::getenv("LLVM_CACHE_REMOTE_SERVICE_SOCKET_PATH");
+
+  if (ServicePath) {
     Args.append({"-Xclang", "-fcompilation-caching-service-path", "-Xclang",
                  ServicePath});
   }
   Args.append({"-greproducible"});
+
+  if (!llvm::sys::Process::GetEnv("CLANG_CACHE_DISABLE_MCCAS") &&
+      !ServicePath) {
+    Args.push_back("-Xclang");
+    Args.push_back("-fcas-backend");
+    if (llvm::sys::Process::GetEnv("CLANG_CACHE_VERIFY_MCCAS")) {
+      Args.push_back("-Xclang");
+      Args.push_back("-fcas-backend-mode=verify");
+    }
+    Args.push_back("-mllvm");
+    Args.push_back("-cas-friendly-debug-info");
```

Comment:
I think I'd rather turn this on in the Darwin toolchain regardless of whether caching is enabled or not.